Add more control to local sky subtraction

tbowers7 opened the pull request:
This PR adds more user control over the _local sky subtraction_ process.  The only new functionality is the ability to turn off Legendre polynomial fitting, otherwise just adding more knobs.

Add the ability to turn off Legendre polynomial fitting in the spatial direction for local sky subtraction, and add a pass-through parameter to control the width of the local sky subtraction window in terms of the FWHM of the object.

This PR:
  * changes SkySubPar parameter `no_poly` --> `no_poly_global` (with conformal changes elsewhere)
  * adds new SkySubPar parameter `no_poly_local` to control whether a legendre polynomial is used in the spatial direction of the local sky subtraction
  * adds new SkySubPar parameter `local_maskwidth` as a pass-through parameter that defaults to the value already defined in `findobj_skymask.objs_in_slit()`.

Updates to docs and Changes will happen after approval of the substance of this PR.

profxj reviewed (approved):
Thanks!

Let's make sure to run the DevSuite

jhennawi commented on pypeit/core/skysub.py:
```diff
@@ -682,6 +682,8 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, thismask,
     no_local_sky : bool, optional
         If True, do not fit local sky model, only object profile and extract optimally
         The objimage will be all zeros.
+    no_poly : bool, optional
```
Make it clear that the polynomial basis here is for the sky.

jhennawi commented on pypeit/spectrographs/gemini_gnirs.py:
```diff
@@ -87,7 +87,7 @@ def default_pypeit_par(cls):
         par['reduce']['skysub']['bspline_spacing'] = 0.8
         par['reduce']['skysub']['global_sky_std']  = False    # Do not perform global sky subtraction for standard stars
-        par['reduce']['skysub']['no_poly'] = True             # Do not use polynomial degree of freedom for global skysub
+        par['reduce']['skysub']['no_poly_global'] = True      # Do not use polynomial degree of freedom for global skysub
```
Hmm, I wonder if this is why we have sky-subtraction problems for GNIRS? I'm confused why we don't do this for other fixed format echelle's but maybe because the GNIRS slit is narrower. Maybe add a note here asking if this is the right behavior as a TODO, but I would not change with investigation.

jhennawi reviewed (approved):
Looks good me. Minor changes requested. Could you maybe however elaborate here or in the Slack what prompted these modifications. Were you getting bad sky-subtraction results somewhere. I'd like to see what those look like for my own edification/education.

tbowers7 committed:
Add more control to local sky subtraction

This commit allows a pass-through parameter to control the width of the local
sky subtraction window in terms of the FWHM of the object.

	modified:   CHANGES.rst
	modified:   pypeit/core/skysub.py
	modified:   pypeit/find_objects.py
	modified:   pypeit/par/pypeitpar.py

tbowers7 commented:
Following on the conversation in Slack, removed the `no_poly_local` option, leaving only the `local_maskwidth` option as the change.

tbowers7 committed:
Doc update + PR comment

kbwestfall committed:
doc update

kbwestfall committed:
release notes

kbwestfall committed:
doc update

kbwestfall committed:
doc update

kbwestfall committed:
readthedocs python version

kbwestfall committed:
readthedocs python version

kbwestfall committed:
doc directory error

kbwestfall committed:
missing specutils dependency for docs

kbwestfall committed:
force failure on sphinx warnings

kbwestfall committed:
Merge pull request #1573 from pypeit/staged

Merges develop into release (1.12.2 tag)

debora-pe reviewed (approved):
looks good to me.

profxj commented:
@tbowers7 resolve the CHANGES conflict and merge.  Thx!

tbowers7 committed:
Merge branch 'develop' into skysub_local_control

tbowers7 committed:
Merge branch 'release' into skysub_local_control
